src/views/home/HueLightsList.vue

```diff
@@ -0,0 +1,314 @@
+          >
+          </CFormSelect>
+          <br />
+        </div>
+
+        <input
+          :disabled="light.on == false"
+          track-by="$index"
+          v-bind:key="light"
+          type="range"
+          id="vol"
+          name="vol"
+          min="0"
+          max="254"
+          v-model="light.bright"
+          style="width: 100%"
+          v-on:input="light.changeBright(id)"
+        />
+      </CCardBody>
+    </CCard>
+  </div>
+</template>
+
+<script>
+/* eslint-disable */
+let   lock_state    = false;
+const group_regex   = /(^[a-zA-Z ]*)([ \d]*\d$)/
+const api_url       = process.env.VUE_APP_HUE_BRIDGE_URL; 
+const hue_app_token = process.env.VUE_APP_HUE_TOKEN;
+const hue_is_color  = new Map([
+  ["LWA004", false],
+  ["LTG002", false],
+  ["LCT015", true],
+  ["LWA009", false],
+]);
+
+class HueLight {
+  constructor(name, on = false, id, bright, model_id) {
+    this.label    = name;
+    this.on       = on;
+    this.value    = id;
+    this.model_id = model_id;
+    this.is_color = hue_is_color.get(this.model_id);
+    this.bright   = bright;
+  }
+
+  switchState(force = false) {
+    if (lock_state === false || force === true) {
+      lock_state = true;
+      this.on = !this.on;
+      fetch("http://" + api_url + "/api/" + hue_app_token + "/lights/" + this.value + "/state", {
+        method: 'PUT',
+        body: JSON.stringify({
+          on: this.on,
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          lock_state = false;
+        })
+    }
+  }
+
+  changeBright(force = false) {
+    if (lock_state === false || force === true) {
+      lock_state = true;
+      fetch("http://" + api_url + "/api/" + hue_app_token + "/lights/" + this.value + "/state", {
+        method: 'PUT',
+        body: JSON.stringify({
+          bri: new Number(this.bright)
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          lock_state = false;
+        })      
+    }
+  }
+
+  updateLight(updatedLight) {
+    this.label    = this.label == updatedLight['name'] ? this.label : updatedLight['name'];
+    this.on       = this.on == updatedLight['state']['on'] ? this.on : updatedLight['state']['on'];
+    this.bright   = this.bright == updatedLight['state']['bri'] ? this.bright : updatedLight['state']['bri'];
+    //update les color
+  }
+}
+
+class HueLightGroup {
+  constructor(name, on = false, bright, model_id) {
+    this.label    = name;
+    this.on       = on;
+    this.value    = "-1";
+    this.model_id = model_id;
+    this.is_color = hue_is_color.get(this.model_id);
+    this.bright   = bright;
+    this.group    = new Array();
+  }
+
+  switchState(force = false) {
+    if (lock_state === false || force === true) {
+      lock_state  = true;
+      this.on     = !this.on;
+      let light   = this.group.find(element => element.value === this.value);
+      light.on    = this.on;
+      if (this.value != -1) {
```

Comment:
attention aux assertions en JS, là ça marche psk en js `!=` veut dire `equivalent` et dans les faits `-1` et `"-1"` sont equivalent dans le sens oú une string `"-1"` convertie en nombre donne `-1` mais tu joues à un jeu dangereux.

Il vaut mieux utiliser les comparaisons strictes avec `!==` plutot que `!=` et de faire en sorte que `this.value` soit déclaré avec pour valeur `-1` et non `"-1"`.

Aussi je suis pas fan du nom de ta variable, `value` ça veut litteralement dire n'importe quoi

Reply:
Ouais j'avais vu ça avec les === et les ==, je fix ça c'était juste un oublie de ma part